src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -422,6 +425,111 @@ public static GeoPoint parseGeoPoint(XContentParser parser, GeoPoint point) thro
         }
     }
 
+    public static boolean correctPolyAmbiguity(GeoPoint[] points, boolean handedness) {
+        return correctPolyAmbiguity(points, handedness, computePolyOrientation(points), 0, 0, points.length, false);
+    }
+
+    public static boolean correctPolyAmbiguity(GeoPoint[] points, boolean handedness, boolean orientation, int component, int offset,
+            int length, boolean shellCorrected) {
+        // OGC requires shell as ccw (Right-Handedness) and holes as cw (Left-Handedness)
+        // since GeoJSON doesn't specify (and doesn't need to) GEO core will assume OGC standards
+        // thus if orientation is computed as cw, the logic will translate points across dateline
+        // and convert to a right handed system
+
+        boolean translated;
+        // compute the bounding box and calculate range
+        Pair<Pair, Pair> range = GeoUtils.computeBBox(points, offset, length);
+        final double rng = (Double)range.getLeft().getRight() - (Double)range.getLeft().getLeft();
+        // translate the points if the following is true
+        //   1.  shell orientation is cw and range is greater than a hemisphere (180 degrees) but not spanning 2 hemispheres
+        //       (translation would result in a collapsed poly)
+        //   2.  the shell of the candidate hole has been translated (to preserve the coordinate system)
+        boolean incorrectOrientation = component == 0 && handedness != orientation;
+        if ( translated = ((incorrectOrientation && (rng > DATELINE && rng != 360.0)) || (shellCorrected && component != 0))) {
```

Comment:
I think it would be cleaner to set translated outside of the if statement.
```
boolean translated = incorrectOrientation && rng > DATELINE && rng != 360.0;
if (translated || shellCorrected && component != 0) {
...
```